cura/Machines/Models/MaterialManagementModel.py

```diff
@@ -0,0 +1,169 @@
+    #   \return Whether or not the material can be removed.
+    @pyqtSlot("QVariant", result = bool)
+    def canMaterialBeRemoved(self, material_node: "MaterialNode"):
+        container_registry = CuraContainerRegistry.getInstance()
+        ids_to_remove = {metadata.get("id", "") for metadata in container_registry.findInstanceContainersMetadata(base_file = material_node.base_file)}
+        for extruder_stack in container_registry.findContainerStacks(type = "extruder_train"):
+            if extruder_stack.material.getId() in ids_to_remove:
+                return False
+        return True
+
+    ##  Change the user-visible name of a material.
+    #   \param material_node The ContainerTree node of the material to rename.
+    #   \param name The new name for the material.
+    @pyqtSlot("QVariant", str)
+    def setMaterialName(self, material_node: "MaterialNode", name: str) -> None:
+        container_registry = CuraContainerRegistry.getInstance()
+        root_material_id = material_node.base_file
+        if container_registry.isReadOnly(root_material_id):
+            Logger.log("w", "Cannot set name of read-only container %s.", root_material_id)
+            return
+        return container_registry.findContainers(id = root_material_id)[0].setName(name)
+
+    ##  Deletes a material from Cura.
+    #
+    #   This function does not do any safety checking any more. Please call this
+    #   function only if:
+    #   - The material is not read-only.
+    #   - The material is not used in any stacks.
+    #   If the material was not lazy-loaded yet, this will fully load the
+    #   container. When removing this material node, all other materials with
+    #   the same base fill will also be removed.
+    #   \param material_node The material to remove.
+    @pyqtSlot("QVariant")
+    def removeMaterial(self, material_node: "MaterialNode") -> None:
+        container_registry = CuraContainerRegistry.getInstance()
+        materials_this_base_file = container_registry.findContainersMetadata(base_file = material_node.base_file)
+        for material_metadata in materials_this_base_file:
+            container_registry.removeContainer(material_metadata["id"])
+
+    ##  Creates a duplicate of a material with the same GUID and base_file
+    #   metadata.
+    #   \param material_node The node representing the material to duplicate.
+    #   \param new_base_id A new material ID for the base material. The IDs of
+    #   the submaterials will be based off this one. If not provided, a material
+    #   ID will be generated automatically.
+    #   \param new_metadata Metadata for the new material. If not provided, this
+    #   will be duplicated from the original material.
+    #   \return The root material ID of the duplicate material.
+    @pyqtSlot("QVariant", result = str)
+    def duplicateMaterial(self, material_node: "MaterialNode", new_base_id: Optional[str] = None, new_metadata: Dict[str, Any] = None) -> Optional[str]:
```

Comment:
I made the `MaterialManager.duplicateMaterial()` function use the `MaterialManagementModel` instead, since this class here is deprecated: 0eb3c5c0d63d61c390c450a1b066fd2821277df4